From 4ed51791e03b63adbaf792564aa201a6d71a1050 Mon Sep 17 00:00:00 2001
From: Barnabas Pocze <pobrn@protonmail.com>
Date: Sat, 18 May 2024 01:04:42 +0200
Subject: [PATCH 6/7] linking: return after aborting transition

Avoid calling return_error() on the same transition multiple times.

Fixes: 4b153ec55354da ("link-target.lua: change into a async hook")
See #628
---
 src/scripts/linking/link-target.lua | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/scripts/linking/link-target.lua b/src/scripts/linking/link-target.lua
index 62be86cec..f49bf513f 100644
--- a/src/scripts/linking/link-target.lua
+++ b/src/scripts/linking/link-target.lua
@@ -50,6 +50,7 @@ AsyncEventHook {
             si_flags.failed_count > 5 then
           transition:return_error ("tried to link on last rescan, not retrying "
               .. tostring (si_link))
+          return
         end
 
         if si_props ["item.factory.name"] == "si-audio-virtual" then
-- 
2.45.1

